sdk/nodejs/src/request.ts

```diff
@@ -43,87 +14,119 @@ const TRACK_PATH = '/v1/track'
 const BUCKETING_URL = `${BUCKETING_BASE}${HOST}`
 const EDGE_DB_QUERY_PARAM = '?enableEdgeDB='
 
+export class ResponseError extends Error {
+    constructor(message: string) {
+        super(message)
+        this.name = 'ResponseError'
+    }
+
+    status: number
+}
+
+const exponentialBackoff: RequestInitWithRetry['retryDelay'] = (
+    attempt
+) => {
+    return Math.pow(2, attempt) * 1000
+}
+
+type retryOnRequestErrorFunc = (retries: number) => RequestInitWithRetry['retryOn']
+
+const retryOnRequestError: retryOnRequestErrorFunc = (retries) => {
+    return (attempt, error, response) => {
+        if (attempt >= retries) {
+            return false
+        }
+
+        if (response && response?.status < 500) {
+            return false
+        }
+
+        return true
+    }
+}
+
+const handleResponse = async (res: Response) => {
+    if (!res.ok) {
+        let error
+        try {
+            error = new ResponseError((await res.clone().json()).message || 'Something went wrong')
+        } catch (e) {
+            error = new ResponseError('Something went wrong')
+        }
+        error.status = res.status
+        throw error
+    }
+
+    return res
+}
+
 export async function publishEvents(
     logger: DVCLogger,
     envKey: string | null,
     eventsBatch: SDKEventBatchRequestBody,
     eventsBaseURLOverride?: string
-): Promise<AxiosResponse> {
+): Promise<Response> {
     if (!envKey) {
         throw new Error('DevCycle is not yet initialized to publish events.')
     }
     const url = eventsBaseURLOverride ? `${eventsBaseURLOverride}${EVENTS_PATH}` : `${EVENT_URL}${HOST}${EVENTS_PATH}`
-    const res = await post({
-        url,
-        data: { batch: eventsBatch }
-    }, envKey, false)
-    if (res.status !== 201) {
-        logger.error(`Error posting events, status: ${res.status}, body: ${res.data?.message}`)
-    } else {
-        logger.debug(`Posted Events, status: ${res.status}, body: ${res.data?.message}`)
-    }
-
-    return res
+    return await post(url, {
```

Comment:
Do we need to set `application/json` headers?

Reply:
it is being set

Reply:
for POST anyway..